### ISSUE TITLE
Can't easily replace collection mapping

### ISSUE THREAD

**Issue by khainke** [yellicode/typescript-extension#3]
Currently, it is next to impossible to introduce a custom collection mapping (instead of `p[]`), since it is not possible to replace the `DefinitionBuilder` and consequently `getFullTypeName` is triggered in basically every important aspect of the writer:

```ts
    private getFullTypeName(typedElement: elements.TypedElement, fallback?: string): string | undefined {
        const typeName = this.typeNameProvider.getTypeName(typedElement) || fallback;
        if (!typeName)
            return; // no type name and no fallback

        if (elements.isMultiplicityElement(typedElement) && typedElement.isMultivalued()) {
            return `${typeName}[]`;  // <===== NO WAY TO REPLACE THIS
        }
        else return typeName;
    }
```

I dont want to rewrite the entire Writer, i just want to introduce custom collection mapping.

**Comment by mdelaat** [comment]
@khainke No, this isn't possible indeed. I looked into your [PR](https://github.com/yellicode/typescript-extension/pull/4) but am not sure if this is the cleanest way to achieve this. What is your use case? What type(s) of collection do you need?

**Comment by khainke** [comment]
Thanks for looking into it. 
Certainly, this would be a language-dependent thing, but most common are the basic collection types array (e.g. `int[]`), list (`List<int>`), set (`Set<int>`) and (more complicated) map (`Map<string, int>`). This is usually not part of UML standards, but also a common critic for UML. This might prove difficult to implement, since commonly these collections can also be used recursively, like `List<Set<Class>>`. In my project i currently created a bit of an ugly workaround with profiles, but it is probably something that requires an extension of the modeling language itself for a "clean" solution.

**Comment by mdelaat** [comment]
Thanks for your explanation, modeling of generic and nested collections looks challenging... Your suggested changes seem to work well though, I will implement them.

Update: published as version 1.13.0.
